c_glib/meson.build

```diff
@@ -47,13 +47,51 @@ data_dir = join_paths(prefix, get_option('datadir'))
 gnome = import('gnome')
 pkgconfig = import('pkgconfig')
 
-root_inc = include_directories('.')
+base_include_directories = [
+  include_directories('.')
+]
 
-arrow_orc_dependency = dependency('arrow-orc', required: false)
+arrow_cpp_build_dir = get_option('arrow_cpp_build_dir')
+arrow_cpp_build_type = get_option('arrow_cpp_build_type')
+if arrow_cpp_build_dir == ''
+  arrow_cpp_build_lib_dir = ''
+else
+  arrow_cpp_build_lib_dir = join_paths(arrow_cpp_build_dir,
+                                       arrow_cpp_build_type.to_lower())
+endif
+
+if arrow_cpp_build_lib_dir == ''
+  arrow = dependency('arrow')
+  have_arrow_orc = dependency('arrow-orc', required: false).found()
+  arrow_gpu = dependency('arrow-gpu', required: false)
+else
+  base_include_directories += [
+    include_directories(join_paths(arrow_cpp_build_dir, 'src')),
```

Comment:
I did the following instruction.

```console
% PKG_CONFIG_PATH=$(brew --prefix libffi)/lib/pkgconfig:$PKG_CONFIG_PATH meson build -Darrow_cpp_build_dir=$PWD/../cpp/debug -Darrow_cpp_build_type=debug
```
I got the following result.

```
% PKG_CONFIG_PATH=$(brew --prefix libffi)/lib/pkgconfig:$PKG_CONFIG_PATH meson build -Darrow_cpp_build_dir=$PWD/../cpp/debug -Darrow_cpp_build_type=debug
The Meson build system
Version: 0.47.2
Source dir: /Users/shiro/arrow/c_glib
Build dir: /Users/shiro/arrow/c_glib/build
Build type: native build
Project name: arrow-glib
Project version: undefined
Native C compiler: ccache cc (clang 9.0.0 "Apple LLVM version 9.0.0 (clang-900.0.38)")
Native C++ compiler: ccache c++ (clang 9.0.0 "Apple LLVM version 9.0.0 (clang-900.0.38)")
Build machine cpu family: x86_64
Build machine cpu: x86_64

meson.build:68:2: ERROR:  Tried to form an absolute path to a source dir. You should not do that but use
relative paths instead.

To get include path to any directory relative to the current dir do

incdir = include_directories(dirname)

After this incdir will contain both the current source dir as well as the
corresponding build dir. It can then be used in any subdirectory and
Meson will take care of all the busywork to make paths work.

Dirname can even be '.' to mark the current directory. Though you should
remember that the current source and build directories are always
put in the include directories by default so you only need to do
include_directories('.') if you intend to use the result in a
different subdirectory.
```

Do we need to specify relative paths on this line? Or is there a mistake in specifying options?

Reply:
We can't use `..` in the path.
Can you try `-Darrow_cpp_build_dir=$(realpath $PWD/../cpp/debug)`? (`cpp/debug` may be a typo of `cpp/build`.)